packages/strapi-design-system/src/RawTable/RawCell.js

```diff
@@ -1,51 +1,151 @@
 
-export const RawTh = ({ isFocusable, ...props }) => {
-  const thRef = useRef(null);
+import { useTable } from './RawTableContext';
 
-  useLayoutEffect(() => {
-    const focusableNodes = getFocusableNodes(thRef.current, true);
-    if (focusableNodes.length === 0) {
-      thRef.current.setAttribute('tabIndex', isFocusable ? 0 : -1);
-    } else {
-      focusableNodes.forEach((node) => node.setAttribute('tabIndex', isFocusable ? 0 : -1));
+export const RawTh = (props) => <RawTd {...props} as="th" />;
+
+export const RawTd = ({ coords, as, ...props }) => {
+  const tdRef = useRef(null);
+  const { rowIndex, colIndex, setTableValues } = useTable();
+  const [isActive, setIsActive] = useState(false);
+
+  /** @type {import("react").KeyboardEventHandler<HTMLTableCellElement> } */
+  const handleKeyDown = (e) => {
+    const focusableNodes = getFocusableNodes(tdRef.current, true);
+
+    /**
+     * If the cell does not have focusable children or if it has focusable children
+     * without keyboard navigation, we should not run the handler.
+     */
+    if (
+      focusableNodes.length === 0 ||
+      (focusableNodes.length === 1 && getFocusableNodesWithKeyboardNav(focusableNodes).length === 0)
+    ) {
+      return;
     }
-  }, [isFocusable]);
 
-  return <Box as="th" ref={thRef} {...props} />;
-};
+    if (e.key === KeyboardKeys.ENTER && !isActive) {
+      setIsActive(true);
+      /**
+       * Cells should be "escapeable" with the escape key or enter key
+       */
+    } else if ((e.key === KeyboardKeys.ESCAPE || e.key === KeyboardKeys.ENTER) && isActive) {
+      setIsActive(false);
+      tdRef.current.focus();
+    } else if (isActive) {
+      /**
+       * This stops the table navigation from working
+       */
+      e.stopPropagation();
+    }
+  };
 
-export const RawTd = ({ isFocusable, ...props }) => {
-  const tdRef = useRef(null);
+  const isFocused = rowIndex === coords.row - 1 && colIndex === coords.col - 1;
 
+  /**
+   * Handles tabindex of the rendered cell element
+   */
   useLayoutEffect(() => {
     const focusableNodes = getFocusableNodes(tdRef.current, true);
 
-    if (focusableNodes.length === 0) {
-      tdRef.current.setAttribute('tabIndex', isFocusable ? 0 : -1);
+    /**
+     * We should focus the cell if there are no focussable children inside
+     * If there is only one focusable child and it has it's own keyboard navigation
+     * Or if there is more than one focusable child.
+     */
+    if (
+      focusableNodes.length === 0 ||
+      (focusableNodes.length === 1 && getFocusableNodesWithKeyboardNav(focusableNodes).length !== 0) ||
+      focusableNodes.length > 1
+    ) {
+      tdRef.current.setAttribute('tabIndex', !isActive && isFocused ? 0 : -1);
+
+      focusableNodes.forEach((node, index) => {
+        node.setAttribute('tabIndex', isActive ? 0 : -1);
+        /**
+         * When a cell is active we want to focus the
+         * first focusable element simulating a focus trap
```

Comment:
to make sure I understand, was your goal to have a focus trap behaviour when entering into a cell with focusable elements?
when entering this cell:
<img width="147" alt="image" src="https://user-images.githubusercontent.com/71838159/192256034-77799230-be80-4845-9951-456e7ee7ed94.png">

after tabbing through all focusable elements:
<img width="154" alt="image" src="https://user-images.githubusercontent.com/71838159/192256138-1ea7cf48-9986-4168-92a6-15621a920d2c.png">

I end up in the Table footer:
<img width="1071" alt="image" src="https://user-images.githubusercontent.com/71838159/192256212-234843cb-773f-438f-a6f5-8ee61fa0b0d4.png">

I don't have a strong opinion on it, I'm able to recover my previous position with `tab reverse` then `Enter`

Reply:
That's the normal behaviour because pressing `tab` will move the focus to the next "focusable" element but I agree that it is a weird behavior in this case. Do you think we could fix it? @joshuaellis

Reply:
That then breaks the aria-pattern, which is sort of the cause of this issue (note point 1)

<img width="981" alt="CleanShot 2022-09-26 at 14 51 02@2x" src="https://user-images.githubusercontent.com/37798644/192293919-c8c659fb-39b3-4573-aeec-5a014310af18.png">

Where we originally would go straight to the `input` but that's not expected behaviour and the keyboard navigation prevents default actions so people can't move the caret.

IMHO whilst we think it's a bit odd for behaviour I do strongly believe we should adhere to the guidelines. I think changing for button is tolerable, but we should draw the line there. 

We don't have control over what people put in their tables anyway so if someone does challenge the implementation then at the very least we have the standards documentation to point too.

Reply:
Link to documentation – https://www.w3.org/WAI/ARIA/apg/patterns/grid/

Reply:
Good and strong points. I'm ok to restrict it to buttons only, thanks @joshuaellis 🔥